lib/libutee/tee_parse_uuid.c

```diff
@@ -0,0 +1,83 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2019, Linaro Limited
+ */
+
+#include <ctype.h>
+#include <stdint.h>
+#include <string.h>
+#include <tee_api_types.h>
+#include <tee_internal_api_extensions.h>
+
+static int hex(char c)
+{
+	char lc = tolower(c);
+
+	if (isdigit(lc))
+		return lc - '0';
+	if (isxdigit(lc))
+		return lc - 'a' + 10;
+	return -1;
+}
+
+static uint32_t parse_hex(const char *s, size_t nchars, uint32_t *res)
+{
+	uint32_t v = 0;
+	size_t n = 0;
+	int c = 0;
+
+	for (n = 0; n < nchars; n++) {
+		c = hex(s[n]);
+		if (c == (char)-1) {
+			*res = TEE_ERROR_BAD_FORMAT;
+			goto out;
+		}
+		v = (v << 4) + c;
+	}
+	*res = TEE_SUCCESS;
+out:
+	return v;
+}
+
+/*
+ * Convert a UUID string @s into a TEE_UUID @uuid
+ * Expected format for @s is: xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx
+ * 'x' being any hexadecimal digit (0-9a-fA-F)
+ */
+TEE_Result tee_parse_uuid(const char *s, TEE_UUID *uuid)
+{
+	TEE_Result res = TEE_SUCCESS;
+	TEE_UUID u = { 0 };
+	const char *p = s;
+	size_t i = 0;
+
+	if (strlen(p) != 36)
```

Comment:
We could be a bit more paranoid, how about:
`!p || strnlen(p, 37) != 36`

Reply:
Sounds good